compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -133,6 +133,10 @@ struct LoweringContext<'a, 'hir: 'a> {
     /// NodeIds that are lowered inside the current HIR owner.
     node_id_to_local_id: FxHashMap<NodeId, hir::ItemLocalId>,
 
+    // The next_node_id is reset for each item.
```

Comment:
I think it would be good to elaborate this comment. It's hard to tell what it means in the current form. Something like:

> Note: We create a new LoweringContext for each item, and each time we initialize `next_node_id` with `resolver.next_node_id`, which means that each time we start with the same value and the id ranges for new nodes created during lowering will overlap. This is fine because these newly created NodeIds don't escape the item's context and are immediately lowered into HirIds using the item-local `node_id_to_local_id` map.